### DISCUSSION TITLE
Heimdall stopped working, update fails

### DISCUSSION THREAD

**Discussion by kvmgz** [tteck/Proxmox#2772] [Help Needed] [answered]
Hi folks,

Heimdall dashboard used to load slowly occasionally, but a restart of the LXC container or even the whole Proxmox host fixed the issue.
Until it didn't. The first sign was the failure of the update of Heimdall using it's script inside the LXC console, giving me this error:
- Updating Heimdall Dashboard to v2.6.1  \
[ERROR] in line 77: exit code 0: while executing command COMPOSER_ALLOW_SUPERUSER=1 composer dump-autoload &> /dev/null
And after this it won't load anymore. 
Sadly I ignored the issue too long, and overwrote my only 3 backups, so I can't roll back anymore.

Any ideas which doesn't include to start over?

**Comment by tteck** [comment]
Had someone yesterday with the same issue.
I just can't replicate the problem
```
            _               _       _ _      ___          _     _                         _ 
  /\  /\___(_)_ __ ___   __| | __ _| | |    /   \__ _ ___| |__ | |__   ___   __ _ _ __ __| |
 / /_/ / _ \ | '_ ` _ \ / _` |/ _` | | |   / /\ / _` / __| '_ \| '_ \ / _ \ / _` | '__/ _` |
/ __  /  __/ | | | | | | (_| | (_| | | |  / /_// (_| \__ \ | | | |_) | (_) | (_| | | | (_| |
\/ /_/ \___|_|_| |_| |_|\__,_|\__,_|_|_| /___,' \__,_|___/_| |_|_.__/ \___/ \__,_|_|  \__,_|
                                                                                            
 ✓ Stopped Heimdall Dashboard
 ✓ Backed up Data
 ✓ Updated Heimdall Dashboard to v2.6.1
 ✓ Restored Data
 ✓ Cleaned
 ✓ Started Heimdall Dashboard
 ✓ Updated Successfully
```
See if executing the following commands in the LXC console gets you going
```
cd ~
cp -R database-backup/* /opt/Heimdall/database
cp -R public-backup/* /opt/Heimdall/public
systemctl start heimdall.service
```

**Reply by kvmgz** [reply]
Btw it happened to an Adguard LXC as well once, but it works now.
Trying your commands now.

**Reply by kvmgz** [reply]
It's still not working, and the update still fails.

**Reply by tteck** [reply, marked as answer]
Here's what the command `COMPOSER_ALLOW_SUPERUSER=1 composer dump-autoload` does on mine.
```
root@heimdall-dashboard:/opt/Heimdall# COMPOSER_ALLOW_SUPERUSER=1 composer dump-autoload
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover

   INFO  Discovering packages.  

  graham-campbell/github ........................................................................................ DONE
  laravel/tinker ................................................................................................ DONE
  laravel/ui .................................................................................................... DONE
  laravelcollective/html ........................................................................................ DONE
  nesbot/carbon ................................................................................................. DONE
  nunomaduro/collision .......................................................................................... DONE
  nunomaduro/termwind ........................................................................................... DONE
  spatie/laravel-ignition ....................................................................................... DONE

Generated optimized autoload files containing 7627 classes
```
You're utilizing Debian 12, correct?

**Reply by kvmgz** [reply]
It looks like it's 11
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"

**Reply by tteck** [reply]
_Could_ be a PHP version issue 🤷🏻‍♂️

**Reply by kvmgz** [reply]
Can I upgrade the Debian somehow?

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/discussions/1498

**Reply by kvmgz** [reply]
The updated worked well.
I updated & upgraded everything.
The update script worked without errors.
AND the dashboard works (i even risk it to say, it loads even faster)

Thanks in identifying and fixing the problem!
